lib/bonuses/IBonusBearer.cpp

```diff
@@ -44,14 +43,14 @@ bool IBonusBearer::hasBonus(const CSelector &selector, const CSelector &limit, c
 	return getBonuses(selector, limit, cachingStr)->size() > 0;
 }
 
-bool IBonusBearer::hasBonusOfType(BonusType type, int subtype) const
+bool IBonusBearer::hasBonusOfType(BonusType type, std::optional<int> subtype) const
```

Comment:
Every statement in here has conditional code.

Proposition: split into two methods instead of using default argument:
```c++
hasBonusOfType(BonusType type, int subtype)
hasBonusOfType(BonusType type)
```

Reply:
done